frontend/src/metabase/parameters/components/ValuesSourceSettings/ValuesSourceSettings.tsx

```diff
@@ -23,79 +21,92 @@ export interface ValuesSourceSettingsProps {
   ) => void;
 }
 
-const ValuesSourceSettings = ({
+export function ValuesSourceSettings({
   parameter,
   onChangeQueryType,
   onChangeSourceSettings,
-}: ValuesSourceSettingsProps): JSX.Element => {
+}: ValuesSourceSettingsProps) {
   const queryType = getQueryType(parameter);
   const [isModalOpened, setIsModalOpened] = useState(false);
 
-  const radioOptions = useMemo(() => {
-    return getRadioOptions({
-      queryType: queryType,
-      onEditClick: () => setIsModalOpened(true),
-      // linked filters only work with connected field sources (metabase#33892)
-      isEditDisabled: hasLinkedFilters(parameter),
-    });
-  }, [queryType, parameter]);
-
-  const handleModalClose = useCallback(() => {
-    setIsModalOpened(false);
-  }, []);
+  // linked filters only work with connected field sources (metabase#33892)
+  const disableEdit = hasLinkedFilters(parameter);
+  const openModal = () => setIsModalOpened(true);
+  const closeModal = () => setIsModalOpened(false);
 
   return (
     <>
-      <Radio
-        value={queryType}
-        options={radioOptions}
-        vertical
-        onChange={onChangeQueryType}
-      />
+      <Radio.Group value={queryType} onChange={onChangeQueryType}>
+        <Stack spacing="xs">
+          <RadioLine
+            ownValue="list"
+            selectedValue={queryType}
+            label={t`Dropdown list`}
+            disableEdit={disableEdit}
+            onEditClick={openModal}
+          />
+          <RadioLine
+            ownValue="search"
+            selectedValue={queryType}
+            label={t`Search box`}
+            disableEdit={disableEdit}
+            onEditClick={openModal}
+          />
+          <RadioLine
+            ownValue="none"
+            selectedValue={queryType}
+            label={t`Input box`}
+            hideEdit
+          />
+        </Stack>
+      </Radio.Group>
       {isModalOpened && (
-        <Modal medium onClose={handleModalClose}>
+        <Modal medium onClose={closeModal}>
           <ValuesSourceModal
             parameter={parameter}
             onSubmit={onChangeSourceSettings}
-            onClose={handleModalClose}
+            onClose={closeModal}
           />
         </Modal>
       )}
     </>
   );
-};
+}
 
 function hasLinkedFilters({ filteringParameters }: Parameter) {
   return filteringParameters != null && filteringParameters.length > 0;
 }
 
-interface RadioLabelProps {
-  title: string;
-  isSelected?: boolean;
-  onEditClick?: () => void;
-  isEditDisabled?: boolean;
-}
-
-const RadioLabel = ({
-  title,
-  isSelected,
+function RadioLine({
+  selectedValue,
+  ownValue,
+  label,
+  disableEdit = false,
+  hideEdit = false,
   onEditClick,
-  isEditDisabled = false,
-}: RadioLabelProps): JSX.Element => {
+}: {
+  selectedValue: ValuesQueryType;
+  ownValue: ValuesQueryType;
+  label: string;
+  disableEdit?: boolean;
+  hideEdit?: boolean;
+  onEditClick?: () => void;
```

Comment:
1. We don't need 2 props for this - lack of `onEditClick` could imply `hideEdit`
2. This component could have been much simpler (we could turn 3 props `disabledEdit` + `hideEdit` + `onEditClick` into 1: `children`) if it used composition instead, i.e.

Without editing capabilities:
```tsx
<RadioLine ownValue="none" selectedValue={queryType} label={t`Input box`} />
```

With editing capabilities:
```tsx
<RadioLine ownValue="search" selectedValue={queryType} label={t`Search box`}>
  {queryType === "search" && (
    <QueryTypeEditButton disabled={disableEdit} onClick={openModal} />
  )}
</RadioLine>
```